tests/functional/ITILValidationTemplate.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace tests\units;
+
+include_once __DIR__ . '/../abstracts/AbstractITILChildTemplate.php';
+
+use ITILValidationTemplate as GlobalITILValidationTemplate;
+use ITILValidationTemplate_Target;
+use tests\units\Glpi\AbstractITILChildTemplate;
+
+class ITILValidationTemplate extends AbstractITILChildTemplate
+{
+    public function testPostTargets()
+    {
+        $validationTemplate = new GlobalITILValidationTemplate();
+        $this->integer((int)$validationTemplate->add([
+            'name'         => 'Validation template',
+            'description'  => 'a description',
+            'content'      => '',
+        ]))->isGreaterThan(0);
+
+        // Add a user target
+        $validationTemplate->input = [
+            'itemtype_target' => 'User',
+            'items_id_target' => 1
+        ];
+
+        $validationTemplate->post_addItem();
+        $targets = ITILValidationTemplate_Target::getTargets($validationTemplate->getID());
+
+        $target = current($targets);
+        $this->integer(count($targets))->isEqualTo(1);
```

Comment:
Before using `current()`, you should assert that `$targets` is an array.
```suggestion
        $targets = ITILValidationTemplate_Target::getTargets($validationTemplate->getID());
        $this->array($targets)->hasSize(1);

        $target = current($targets);
```